src/preconditioning.jl

```diff
@@ -15,10 +14,7 @@ end
 @fastmath @inbounds function make_preconditioner!(P, h, d, q)
     val = factorial(q) / h^(q + 1 / 2)
     for j in 0:q
-        @simd ivdep for i in 0:d-1
-            # P[j+i*(q+1)+1, j+i*(q+1)+1] = val
-            P.diag[j+i*(q+1)+1] = val
-        end
+        P.B.diag[j+1] = val
```

Comment:
We shouldn't actually rely on the field name. Figure out the best way to access the right field instead.